pkg/reconciler/ksvc/resources/knative_route.go

```diff
@@ -31,15 +31,27 @@ func CreateKnativeRoute(kfsvc *v1alpha1.KFService) *knservingv1alpha1.Route {
 			},
 		})
 	}
+	var kfsvcAnnotations map[string]string
+	filteredAnnotations := filter(kfsvc.Annotations, isFilteredRouteAnnotation)
+	if len(filteredAnnotations) > 0 {
+		kfsvcAnnotations = filteredAnnotations
+	}
 	return &knservingv1alpha1.Route{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:        kfsvc.Name,
 			Namespace:   kfsvc.Namespace,
 			Labels:      kfsvc.Labels,
-			Annotations: kfsvc.Annotations,
+			Annotations: kfsvcAnnotations,
 		},
 		Spec: knservingv1alpha1.RouteSpec{
 			Traffic: trafficTargets,
 		},
 	}
 }
+
+func isFilteredRouteAnnotation(annotationKey string) bool {
```

Comment:
Not sure if you like the pattern, but I'm familiar with naming filters as:

routeAnnotationFilter